x-pack/legacy/plugins/siem/public/components/alerts_viewer/histogram_configs.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import * as i18n from './translations';
+import { MatrixHistogramOption } from '../matrix_histogram/types';
+import { HistogramType } from '../../graphql/types';
+
+export const alertsStackByOptions: MatrixHistogramOption[] = [
+  {
+    text: 'event.category',
+    value: 'event.category',
+  },
+  {
+    text: 'event.module',
+    value: 'event.module',
+  },
+];
+
+export const histogramConfigs = {
+  defaultStackByOption: alertsStackByOptions[1],
+  errorMessage: i18n.ERROR_FETCHING_ALERTS_DATA,
+  histogramType: HistogramType.alerts,
```

Comment:
I like how this configuration object groups all the related properties of a histogram that are commonly customized. Out of curiosity, is there much branching logic remaining on `HistogramType`, and if not, is it now possible to eliminate this property / touchpoint?

Reply:
We still rely on the histogramType for server side to check what kind of configs to apply https://github.com/elastic/kibana/blob/e631eef33dcac85ed5a0f80258fe55e90a569e25/x-pack/legacy/plugins/siem/server/lib/matrix_histogram/elasticsearch_adapter.ts#L120.